### PR TITLE
Implement build-check to ensure safe merges

### PR DESCRIPTION
**Pull request by ncksllvn**
This pull request implements a Jenkins step to build the FE project with the content contained in each pull request. If the build fails, va-bot will comment with the output. The output is not entirely friendly for content editors, but it's very useful for engineers and it was simple to parse.

You can see a sample comment in this PR.

Resolves https://github.com/department-of-veterans-affairs/vets.gov-team/issues/15021

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -11,8 +46,38 @@ node('vetsgov-general-purpose') {
...
+    }
+
+    def output = sh(returnStdout: true, script: "cat output.log").trim()
+    def hasError = output.indexOf('npm ERR!') > -1
+    if (hasError) {
+      commentBrokenLinks(output)
+      error(output)
+    }
+  }
+
```
Comment (ncksllvn) [inline Jenkinsfile]: There's obviously a lot of code redundancy here with vets-website's Jenkinsfile, but I think we can address that when we implement the content deployment.

**Comment by ncksllvn** [comment]
If someone has a moment, would love a review on this so I can stop stressing about `vagov-content` breaking the build 😬

File: Jenkinsfile [hunk 2]
```diff
@@ -1,8 +1,43 @@
...
+APP_CODE_REPO = 'vets-website'
+CONTENT_REPO = 'vagov-content'
+
+def checkoutAppCode = {
+  final scmOptions = [
+    $class: 'GitSCM',
+    branches: [[name: '*/master']],
+    doGenerateSubmoduleConfigurations: false,
+    extensions: [
+      [$class: 'CloneOption', noTags: false, reference: '', shallow: true]
```
Comment (jkassemi) [inline Jenkinsfile]: is `noTags: false` a requirement here?

**Comment by ncksllvn** [comment]
Unless the workflow of the content team changes based on having a dedicated repo, there shouldn't be any increases to deployment times, because each PR merged to `master` here just rebuilds `master` on `vets-website` which is effectively the same as it was prior to splitting the content out. This deployment step is actually in effect already. The build step implemented here is purely a build - no archive or publish - so unless that affects the CI resources, then I don't think there should be any effect on deployments.

File: Jenkinsfile (hunk 2 above)
Comment (ncksllvn) [reply Jenkinsfile]: Good question! I'll try without it 🙂

**Pull request merged**